xwiki-platform-core/xwiki-platform-attachment/xwiki-platform-attachment-validation/xwiki-platform-attachment-validation-api/src/main/java/org/xwiki/attachment/validation/AttachmentValidationScriptService.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.attachment.validation;
+
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.script.service.ScriptService;
+import org.xwiki.stability.Unstable;
+
+/**
+ * Script service for the attachment validation. Provide the attachment validation configuration values.
+ *
+ * @version $Id$
+ * @since 14.10RC1
+ */
+@Component
+@Singleton
+@Named("attachment-validation")
```

Comment:
I don't recall having see `-` used in our script services names? We generally use a `.` as separator.

Reply:
I've used this name because using `attachment.validation` is trying to find `validation` is `AttachmentScriptService`.
The other option is to add a `get` method in `AttachmentScriptService` with a lookup for the "sub service" as we do for instance in `org.xwiki.export.script.ExportScriptService#get`

Reply:
Yes `.` is a very special separator and it needs to make sense as a sub service (and I guess it does here since we have an `attachment` script service).

Reply:
Renamed `attachmentValidation`.